crates/engine/tree/src/tree/mod.rs

```diff
@@ -24,8 +25,7 @@ use reth_payload_builder::PayloadBuilderHandle;
 use reth_payload_primitives::{PayloadAttributes, PayloadBuilderAttributes};
 use reth_payload_validator::ExecutionPayloadValidator;
 use reth_primitives::{
-    Block, BlockNumHash, BlockNumber, GotExpected, Header, SealedBlock, SealedBlockWithSenders,
-    SealedHeader, B256, U256,
+    Block, BlockNumHash, GotExpected, Header, SealedBlock, SealedBlockWithSenders, SealedHeader,
```

Comment:
`BlockNumHash` is in `alloy-eips`, would be great to replace that too

Reply:
Yes, you are correct. However, I aim to limit to only `alloy_primitives` given not too many changes. Let me check if there aren't not too many instance.